### PR TITLE
Develop 🔀 master

### PR DESCRIPTION
**Pull request by pontuslaestadius**
If you find any issues resolve them yourself since everyone is a contributor to this.

### CODE REVIEW

File: .CHAIN_HELP [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Run with: bash chain.sh [ OPTIONS ]
+It is prefered to delete your build directory before running.
+
+	  --pull	  pulls the latest docker images from the docker hub. 
+          --push	  pushes the to the docker hub of the next argument user id.
```
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: So this is a two argument command?

File: chain.sh [hunk 2]
```diff
@@ -13,93 +13,126 @@
...
+        ;;
+	--run)
 
+			# If a container already exists and is running.
+			if [ "$(docker ps -aq -f status=running -f name=$DOCKER_NAME)" ]; then
+
+				# Without force, prompt the user with a choice.
+				if [ "${FORCE}" = "no" ];
+					then
+						# clean up with prompt, only works with bash and often not user shell.
```
Comment (lainqvrnstrm) [inline chain.sh]: Could be worth mentioning in the README?

**Comment by pontuslaestadius** [comment]
Travis says this does not work? why

File: .CHAIN_HELP [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+	  --debug	  runs in debug mode using valgrind.
+-h        --help          displays usages.
+-b        --build         builds using the default Dockerfile in current directory.
```
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: Preferably, add "an image", so we know what it is building.

**Review by lainqvrnstrm — CHANGES_REQUESTED** [review]

File: .CHAIN_HELP (hunk 1 above)
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: "requested access to the resource is denied", is that what we want?

File: .CHAIN_HELP [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Run with: bash chain.sh [ OPTIONS ]
+It is prefered to delete your build directory before running.
+
+	  --pull	  pulls the latest docker images from the docker hub. 
+          --push	  pushes the to the docker hub of the next argument user id.
+	  --debug	  runs in debug mode using valgrind.
```
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: Doesn't work without the build directory, which you recommend to remove above.

Comment (lainqvrnstrm) [reply .CHAIN_HELP]: Does work after having used the --local flag and the project has been built.

File: .CHAIN_HELP [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+-h        --help          displays usages.
+-b        --build         builds using the default Dockerfile in current directory.
+-f        --force         deletes ignores warnings and removes previous container.
+-l        --local         runs without docker container and without ARM compilation
+-d        --delete        deletes the toolchain container.
```
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: "Error: No such container: carpet_compile"

File: chain.sh [hunk 6]
```diff
@@ -13,93 +13,126 @@
...
-    	;;
+	-t|--transfer)
+	
+			docker save -o $DOCKER_NAME.tar $DOCKER_NAME:armhf
+			scp $DOCKER_NAME.tar debian@192.168.8.1:~/
+			rm $DOCKER_NAME.tar
+			ssh debian@192.168.8.1 -t 'sh run.sh'
+	
+        ;;
+	--run)
```
Comment (lainqvrnstrm) [inline chain.sh]: --run flag does not exist in the .CHAIN_HELP file

File: .CHAIN_HELP [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Run with: bash chain.sh [ OPTIONS ]
+It is prefered to delete your build directory before running.
+
+	  --pull	  pulls the latest docker images from the docker hub. 
+          --push	  pushes the to the docker hub of the next argument user id.
+	  --debug	  runs in debug mode using valgrind.
+-h        --help          displays usages.
+-b        --build         builds using the default Dockerfile in current directory.
+-f        --force         deletes ignores warnings and removes previous container.
```
Comment (lainqvrnstrm) [inline .CHAIN_HELP]: Has to be used with --run

File: .CHAIN_HELP (hunk 5 above)
Comment (lainqvrnstrm) [reply .CHAIN_HELP]: Have to run "bash chain.sh --run" first.

Comment (lainqvrnstrm) [reply .CHAIN_HELP]: Should it still be called the toolchain container?

**Review by lainqvrnstrm — CHANGES_REQUESTED** [review]

**Review by lainqvrnstrm — APPROVED** [review]

**Pull request merged**